comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -229,22 +253,21 @@ func (r *secretResolver) resolve(data []byte, origin string, notifyCb secrets.Re
 	haveSecret := false
 
 	w := &walker{
-		resolver: func(yamlPath []string, value string) (string, error) {
+		resolver: func(path []string, value string) (string, error) {
```

Comment:
I think I'm misunderstanding a use case here. I know there's specific needs for the Config: it wants to get callbacks with fine-grained information about what secrets are resolved the first time. Right now this is handled by Resolve() ignoring the allowlist and sending every change. Is this ever going to happen while the cache has values in it? Is that what you mean by "used in a new configuration"?

Reply:
Okay, added code here to update subscriptions when a cached value is found. I think it makes sense: the cache is for efficiency (pertaining to what gets fetched from the backend command) but shouldn't affect subscriptions as far as `Resolve` is concerned, since subscribers need to know about the initial resolution state.